src/unix/core.c

```diff
@@ -998,6 +998,41 @@ int uv_getrusage(uv_rusage_t* rusage) {
   return 0;
 }
 
+int uv_getrusage_thread(uv_rusage_t* rusage) {
+#ifndef RUSAGE_THREAD
```

Comment:
What about having an internal `int uv__getrusage(int who, uv_rusage_t* rusage)` to be called from `uv_getrusage()` and `uv_getrusage_thread()` which implements the actual functionality?

Reply:
For the non-windows case, that makes sense. There's enough difference with the Windows case that having separate impls is better